mm2src/mm2_core/src/mm_ctx.rs

```diff
@@ -324,6 +324,94 @@ impl MmCtx {
 
+    /// Returns the path to the global common directory.
+    ///
+    /// Such directory isn't bound to a specific seed/wallet or address.
+    /// Data that should be stored there is public and shared between all seeds and addresses (e.g. stats, block headers, etc...).
+    #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+    pub fn global_dir(&self) -> Result<PathBuf, String> {
+        let path = path_to_db_root(self.conf["dbdir"].as_str()).join("global");
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| format!("Failed to create global directory: {}", e))?;
+        }
+        Ok(path)
+    }
+
+    /// Returns the path to wallet's data directory.
+    ///
+    /// This path depends on `self.rmd160()` of the wallet derived from the seed.
+    /// For HD wallets, this `rmd160` is derived from `mm2_internal_derivation_path`.
+    /// For Iguana, this `rmd160` is simply a hash of the seed.
+    /// Use this directory to store seed/wallet related data rather than address related data (e.g. HD wallet accounts, HD wallet tx history, etc...)
+    #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+    pub fn wallet_dir(&self) -> Result<PathBuf, String> {
+        let path = path_to_db_root(self.conf["dbdir"].as_str())
+            .join("wallets")
+            .join(hex::encode(self.rmd160().as_slice()));
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| format!("Failed to create wallet directory: {}", e))?;
+        }
+        Ok(path)
+    }
+
+    /// Returns the path to the provided address' data directory.
+    ///
+    /// Use this directory for data related to a specific address and only that specific address (e.g. swap data, order data, etc...).
+    /// This makes sure that when this address is activated using a different technique, this data is still accessible.
+    #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+    pub fn address_dir(&self, address: &str) -> Result<PathBuf, String> {
+        let path = path_to_db_root(self.conf["dbdir"].as_str())
+            .join("addresses")
+            .join(address);
+        if !path.exists() {
+            std::fs::create_dir_all(&path).map_err(|e| format!("Failed to create address directory: {}", e))?;
+        }
+        Ok(path)
+    }
+
+    /// Returns a SQL connection to the global database.
+    #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+    pub fn global_db(&self) -> Result<Connection, String> {
+        // TODO: Initialize this DB on startup and cache it in the context to avoid having this function return `Result`.
+        //       Note that you are using `unwrap`s in the calls to `global_db`.
+        let path = self.global_dir()?.join("global.db");
+        log_sqlite_file_open_attempt(&path);
+        let connection = try_s!(Connection::open(path));
+        Ok(connection)
+    }
```

Comment:
> / TODO: Initialize this DB on startup and cache it in the context to avoid having this function return `Result`.
        //       Note that you are using `unwrap`s in the calls to `global_db`.

in addition to doing this, it will help us avoid performing the operations every time the function is called. There similar functions too e.g wallet_dir, async_wallet_db

Reply:
[01b7085](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2378/commits/01b7085a5d0db945f45fcfb1f71c71ccb084503d)